### PR TITLE
Add overlay for OnePlus 10R

### PR DESCRIPTION
**Pull request by odys2061**
tested successfully in lineage-20.0-20230115-UNOFFICIAL-arm64_bvS.img

### CODE REVIEW

File: OnePlus/OP10R/res/xml/power_profile.xml [hunk 1]
```diff
@@ -0,0 +1,69 @@
+<?xml version="1.0" encoding="utf-8"?>
+<device name="Android">
+    <item name="ambient.on.display0">0.1</item>
+    <item name="screen.on.display0">0.1</item>
+    <item name="screen.full.display0">0.1</item>
+    <item name="bluetooth.active">0.1</item>
```
Comment (phhusson) [inline OnePlus/OP10R/res/xml/power_profile.xml]: If your screen consumed only 0.1mA, your smartphone would have 1 year of battery life screen on.
Is it the case?
I'm guessing not. That means this power profile is wrong.

**Comment by phhusson** [comment]
Where'd you get that power profile? You extracted it from stock rom?
It doesn't look obviously wrong, though at least battery capacity is wrong

**Comment by odys2061** [comment]
> Where'd you get that power profile? You extracted it from stock rom? It doesn't look obviously wrong, though at least battery capacity is wrong

from framework-res.apk of stock Oxygen 13 C.09

**Comment by odys2061** [comment]
OK i will change it

**Comment by phantastar** [comment]
> Where'd you get that power profile? You extracted it from stock rom? It doesn't look obviously wrong, though at least battery capacity is wrong

you mean oneplus purposely reduce battery capacity in latest version?

**Comment by phhusson** [comment]
Where did you try to get power_profile yet?
Maybe the correct one is yet again somewhere else

**Comment by odys2061** [comment]
> Where did you try to get power_profile yet? Maybe the correct one is yet again somewhere else

I searched the whole stock directory using root explorer and get 7 apk about "framework", they are: 
1. android_framework_res_rro_export_oneplus.apk
`(no power_profile.xml)`
2. oplus_framework_res_rro_oneplus.apk 
`(no power_profile.xml)`
3. FrameworkResOverlayExt.apk  
`(no power_profile.xml)`
4. framework-res.apk 
`(power_profile.xml & power_profile_test.xml)`
5. framework-res__auto_generated_rro_product.apk 
`(power_profile.xml)`
6. FrameworkResOverlay.apk 
`(power_profile.xml)`
7. oplus-framework-res.apk 
`(power_profile_13077.xml & power_profile_13097.xml & power_profile_14001.xml)`



Hence I've got 4 power_profile.xml, they are:
1. framework-res.apk > res > xml > power_profile.xml (The original commit, screen consumes only 0.1mA);
```
<?xml version="1.0" encoding="utf-8"?>
<device name="Android">
    <item name="ambient.on.display0">0.1</item>
    <item name="screen.on.display0">0.1</item>
    <item name="screen.full.display0">0.1</item>
    <item name="bluetooth.active">0.1</item>
    <item name="bluetooth.on">0.1</item>
    <item name="wifi.on">0.1</item>
    <item name="wifi.active">0.1</item>
    <item name="wifi.scan">0.1</item>
    <item name="audio">0.1</item>
    <item name="video">0.1</item>
    <item name="camera.flashlight">0.1</item>
    <item name="camera.avg">0.1</item>
    <item name="gps.on">0.1</item>
    <item name="radio.active">0.1</item>
    <item name="radio.scanning">0.1</item>
    <array name="radio.on">
        <value>0.2</value>
        <value>0.1</value>
    </array>
    <array name="cpu.active">
        <value>0.1</value>
    </array>
    <array name="cpu.clusters.cores">
        <value>1</value>
    </array>
    <array name="cpu.speeds.cluster0">
        <value>400000</value>
    </array>
    <array name="cpu.active.cluster0">
        <value>0.1</value>
    </array>
    <item name="cpu.idle">0.1</item>
    <array name="memory.bandwidths">
        <value>22.7</value>
    </array>
    <item name="battery.capacity">1000</item>
    <item name="wifi.controller.idle">0</item>
    <item name="wifi.controller.rx">0</item>
    <item name="wifi.controller.tx">0</item>
    <array name="wifi.controller.tx_levels" />
    <item name="wifi.controller.voltage">0</item>
    <array name="wifi.batchedscan">
        <value>.0002</value>
        <value>.002</value>
        <value>.02</value>
        <value>.2</value>
        <value>2</value>
    </array>
    <modem>
        <sleep>0</sleep>
        <idle>0</idle>
        <active rat="DEFAULT">
            <receive>0</receive>
            <transmit level="0">0</transmit>
            <transmit level="1">0</transmit>
            <transmit level="2">0</transmit>
            <transmit level="3">0</transmit>
            <transmit level="4">0</transmit>
        </active>
    </modem>
    <item name="modem.controller.voltage">0</item>
    <array name="gps.signalqualitybased">
        <value>0</value>
        <value>0</value>
    </array>
    <item name="gps.voltage">0</item>
</device>
```
2. framework-res.apk > res > xml > power_profile_test.xml (The altered commit,  battery capacity is wrong);
```
<?xml version="1.0" encoding="utf-8"?>
<device name="Android">
    <item name="battery.capacity">3000</item>
    <array name="cpu.clusters.cores">
        <value>4</value>
        <value>4</value>
    </array>
    <item name="cpu.suspend">5</item>
    <item name="cpu.idle">1.11</item>
    <item name="cpu.active">2.55</item>
    <item name="cpu.cluster_power.cluster0">2.11</item>
    <item name="cpu.cluster_power.cluster1">2.22</item>
    <array name="cpu.core_speeds.cluster0">
        <value>300000</value>
        <value>1000000</value>
        <value>2000000</value>
    </array>
    <array name="cpu.core_speeds.cluster1">
        <value>300000</value>
        <value>1000000</value>
        <value>2500000</value>
        <value>3000000</value>
    </array>
    <array name="cpu.core_power.cluster0">
        <value>10</value>
        <value>20</value>
        <value>30</value>
    </array>
    <array name="cpu.core_power.cluster1">
        <value>25</value>
        <value>35</value>
        <value>50</value>
        <value>60</value>
    </array>
    <item name="ambient.on">0.5</item>
    <item name="screen.on">100</item>
    <item name="screen.full">800</item>
    <item name="camera.flashlight">500</item>
    <item name="camera.avg">600</item>
    <item name="audio">100.0</item>
    <item name="video">150.0</item>
    <item name="gps.on">10</item>
    <item name="radio.active">60</item>
    <item name="radio.scanning">3</item>
    <array name="radio.on">
        <value>6</value>
        <value>5</value>
        <value>4</value>
        <value>3</value>
        <value>3</value>
    </array>
</device>
```
3. framework-res__auto_generated_rro_product.apk  > res > xml > power_profile.xml (screen consumes only 0.1mA);
```
<?xml version="1.0" encoding="utf-8"?>
<device name="Android">
    <item name="ambient.on">0.1</item>
    <item name="screen.on">0.1</item>
    <item name="screen.full">0.1</item>
    <item name="bluetooth.active">0.1</item>
    <item name="bluetooth.on">0.1</item>
    <item name="wifi.on">0.1</item>
    <item name="wifi.active">0.1</item>
    <item name="wifi.scan">0.1</item>
    <item name="audio">0.1</item>
    <item name="video">0.1</item>
    <item name="camera.flashlight">0.1</item>
    <item name="camera.avg">0.1</item>
    <item name="gps.on">0.1</item>
    <item name="radio.active">0.1</item>
    <item name="radio.scanning">0.1</item>
    <array name="radio.on">
        <value>0.2</value>
        <value>0.1</value>
    </array>
    <array name="cpu.clusters.cores">
        <value>1</value>
    </array>
    <array name="cpu.speeds.cluster0">
        <value>400000</value>
    </array>
    <array name="cpu.active.cluster0">
        <value>0.1</value>
    </array>
    <item name="cpu.idle">0.1</item>
    <item name="cpu.suspend">5</item>
    <item name="cpu.active">2.55</item>. <item name="cpu.cluster_power.cluster0">2.11</item>
    <item name="cpu.cluster_power.cluster1">2.22</item>
    <array name="memory.bandwidths">
        <value>22.7</value>
    </array>
    <item name="battery.capacity">1000</item>
    <item name="wifi.controller.idle">0</item>
    <item name="wifi.controller.rx">0</item>
    <item name="wifi.controller.tx">0</item>
    <array name="wifi.controller.tx_levels" />
    <item name="wifi.controller.voltage">0</item>
    <array name="wifi.batchedscan">
        <value>.0002</value>
        <value>.002</value>
        <value>.02</value>
        <value>.2</value>
        <value>2</value>
    </array>
    <item name="modem.controller.sleep">0</item>
    <item name="modem.controller.idle">0</item>
    <item name="modem.controller.rx">0</item>
    <array name="modem.controller.tx">
        <value>0</value>
        <value>0</value>
        <value>0</value>
        <value>0</value>
        <value>0</value>
    </array>
    <item name="modem.controller.voltage">0</item>
    <array name="gps.signalqualitybased">
        <value>0</value>
        <value>0</value>
    </array>
    <item name="gps.voltage">0</item>
</device>
```
4. FrameworkResOverlay.apk > res > xml > power_profile.xml (screen consumes only 0.1mA);
```
<?xml version="1.0" encoding="utf-8"?>
<device name="Android">
    <item name="ambient.on">0.1</item>
    <item name="screen.on">0.1</item>
    <item name="screen.full">0.1</item>
    <item name="bluetooth.active">0.1</item>
    <item name="bluetooth.on">0.1</item>
    <item name="wifi.on">0.1</item>
    <item name="wifi.active">0.1</item>
    <item name="wifi.scan">0.1</item>
    <item name="audio">0.1</item>
    <item name="video">0.1</item>
    <item name="camera.flashlight">0.1</item>
    <item name="camera.avg">0.1</item>
    <item name="gps.on">0.1</item>
    <item name="radio.active">0.1</item>
    <item name="radio.scanning">0.1</item>
    <array name="radio.on">
        <value>0.2</value>
        <value>0.1</value>
    </array>
    <array name="cpu.clusters.cores">
        <value>4</value>
        <value>4</value>
    </array>
    <array name="cpu.speeds.cluster0">
        <value>400000</value>
    </array>
    <array name="cpu.active.cluster0">
        <value>0.1</value>
    </array>
    <item name="cpu.idle">1.11</item>
    <item name="cpu.suspend">5</item>
    <item name="cpu.active">2.55</item>. <item name="cpu.cluster_power.cluster0">2.11</item>
    <item name="cpu.cluster_power.cluster1">2.22</item>
    <array name="cpu.core_speeds.cluster0">
        <value>300000</value>
        <value>1000000</value>
        <value>2000000</value>
    </array>
    <array name="cpu.core_speeds.cluster1">
        <value>300000</value>
        <value>1000000</value>
        <value>2500000</value>
        <value>3000000</value>
    </array>
    <array name="cpu.core_power.cluster0">
        <value>10</value>
        <value>20</value>
        <value>30</value>
    </array>
    <array name="cpu.core_power.cluster1">
        <value>25</value>
        <value>35</value>
        <value>50</value>
        <value>60</value>
    </array>
    <array name="memory.bandwidths">
        <value>22.7</value>
    </array>
    <item name="battery.capacity">1000</item>
    <item name="wifi.controller.idle">0</item>
    <item name="wifi.controller.rx">0</item>
    <item name="wifi.controller.tx">0</item>
    <array name="wifi.controller.tx_levels" />
    <item name="wifi.controller.voltage">0</item>
    <array name="wifi.batchedscan">
        <value>.0002</value>
        <value>.002</value>
        <value>.02</value>
        <value>.2</value>
        <value>2</value>
    </array>
    <item name="modem.controller.sleep">0</item>
    <item name="modem.controller.idle">0</item>
    <item name="modem.controller.rx">0</item>
    <array name="modem.controller.tx">
        <value>0</value>
        <value>0</value>
        <value>0</value>
        <value>0</value>
        <value>0</value>
    </array>
    <item name="modem.controller.voltage">0</item>
    <array name="gps.signalqualitybased">
        <value>0</value>
        <value>0</value>
    </array>
    <item name="gps.voltage">0</item>
</device>
```
5. oplus-framework-res.apk > res > xml > power_profile_13077.xml (battery capacity is wrong);
```
<?xml version="1.0" encoding="utf-8"?>
<device name="Android">
    <item name="none">0</item>
    <item name="screen.on">79.6</item>
    <item name="screen.full">232.2</item>
    <item name="bluetooth.active">47</item>
    <item name="bluetooth.on">0.9</item>
    <item name="wifi.on">1.6</item>
    <item name="wifi.active">132.5</item>
    <item name="wifi.scan">172</item>
    <item name="dsp.audio">81</item>
    <item name="dsp.video">152</item>
    <item name="radio.active">125.5</item>
    <item name="radio.scanning">2</item>
    <item name="gps.on">65.5</item>
    <array name="radio.on">
        <value>2</value>
        <value>1</value>
    </array>
    <array name="cpu.speeds">
        <value>300000</value>
        <value>422400</value>
        <value>652800</value>
        <value>729600</value>
        <value>883200</value>
        <value>960000</value>
        <value>1036800</value>
        <value>1190400</value>
        <value>1267200</value>
        <value>1497600</value>
        <value>1574400</value>
        <value>1728000</value>
        <value>1958400</value>
        <value>2265600</value>
    </array>
    <item name="cpu.idle">2.8</item>
    <array name="cpu.active">
        <value>75</value>
        <value>104</value>
        <value>132</value>
        <value>136.5</value>
        <value>143.9</value>
        <value>178</value>
        <value>184</value>
        <value>190</value>
        <value>246</value>
        <value>268</value>
        <value>298</value>
        <value>358</value>
        <value>378</value>
        <value>420</value>
    </array>
    <item name="battery.capacity">2800</item>
</device>
```
6. oplus-framework-res.apk > res > xml > power_profile_13097.xml (battery capacity is wrong);
```
<?xml version="1.0" encoding="utf-8"?>
<device name="Android">
    <item name="none">0</item>
    <item name="screen.on">180.3</item>
    <item name="screen.full">240.2</item>
    <item name="bluetooth.active">47</item>
    <item name="bluetooth.on">0.9</item>
    <item name="wifi.on">1.6</item>
    <item name="wifi.active">132.5</item>
    <item name="wifi.scan">172</item>
    <item name="dsp.audio">81</item>
    <item name="dsp.video">152</item>
    <item name="radio.active">125.5</item>
    <item name="radio.scanning">2</item>
    <item name="gps.on">65.5</item>
    <array name="radio.on">
        <value>2</value>
        <value>1</value>
    </array>
    <array name="cpu.speeds">
        <value>300000</value>
        <value>422400</value>
        <value>652800</value>
        <value>729600</value>
        <value>883200</value>
        <value>960000</value>
        <value>1036800</value>
        <value>1190400</value>
        <value>1267200</value>
        <value>1497600</value>
        <value>1574400</value>
        <value>1728000</value>
        <value>1958400</value>
        <value>2265600</value>
        <value>2457600 </value>
    </array>
    <item name="cpu.idle">4.3</item>
    <array name="cpu.active">
        <value>75</value>
        <value>104</value>
        <value>132</value>
        <value>136.5</value>
        <value>143.9</value>
        <value>178</value>
        <value>184</value>
        <value>190</value>
        <value>246</value>
        <value>268</value>
        <value>298</value>
        <value>358</value>
        <value>378</value>
        <value>420</value>
        <value>430</value>
    </array>
    <item name="battery.capacity">3100</item>
</device>
```
7. oplus-framework-res.apk > res > xml > power_profile_14001.xml (battery capacity is wrong);
```
<?xml version="1.0" encoding="utf-8"?>
<device name="Android">
    <item name="none">0</item>
    <item name="screen.on">79.6</item>
    <item name="screen.full">232.2</item>
    <item name="bluetooth.active">47</item>
    <item name="bluetooth.on">0.9</item>
    <item name="wifi.on">1.6</item>
    <item name="wifi.active">132.5</item>
    <item name="wifi.scan">172</item>
    <item name="dsp.audio">81</item>
    <item name="dsp.video">152</item>
    <item name="radio.active">125.5</item>
    <item name="radio.scanning">2</item>
    <item name="gps.on">65.5</item>
    <array name="radio.on">
        <value>2</value>
        <value>1</value>
    </array>
    <array name="cpu.speeds">
        <value>300000</value>
        <value>422400</value>
        <value>652800</value>
        <value>729600</value>
        <value>883200</value>
        <value>960000</value>
        <value>1036800</value>
        <value>1190400</value>
        <value>1267200</value>
        <value>1497600</value>
        <value>1574400</value>
        <value>1728000</value>
        <value>1958400</value>
        <value>2265600</value>
        <value>2457600 </value>
    </array>
    <item name="cpu.idle">4.7</item>
    <array name="cpu.active">
        <value>75</value>
        <value>104</value>
        <value>132</value>
        <value>136.5</value>
        <value>143.9</value>
        <value>178</value>
        <value>184</value>
        <value>190</value>
        <value>246</value>
        <value>268</value>
        <value>298</value>
        <value>358</value>
        <value>378</value>
        <value>420</value>
        <value>430</value>
    </array>
    <item name="battery.capacity">3100</item>
</device>
```

**Comment by odys2061** [comment]
IMO I think it's better to choose power_profile_test.xml from framework-res.apk (and manually change the battery capacity to real one), because I found the xml files from 10 pro and Ace Pro are also lower than real value on battery capacity. It might be a common problem?

10 Pro:
1. https://www.oneplus.com/us/10-pro/specs ("`Battery: 5,000 mAh (2S1P 2,500 mAh, non-removable`")
2. from 10 Pro's xml file: `<item name="battery.capacity">4040</item>`

**Comment by phhusson** [comment]
Check your prj name/id `adb shell 'getprop |grep prj` and see which power_profile_<number>.xml it matches, I'm guessing it's rather one of those that you should use, and i'm guessing that oppo changed framework to actually read battery capacity rather than using this battery.capacity value

**Comment by odys2061** [comment]
> getprop |grep prj

Does that mean I haven't got the correct power_profile.xml? (But I've extracted all apks about "framework")

```
OP5566L1:/ $ getprop |grep prj
[ro.boot.prjname]: [21861]
```

**Comment by phhusson** [comment]
power_profile_test is definitely wrong, it's coming from AOSP.

I would go with power_profile_13077 which has reasonable screen.on consumption

**Pull request closed without merging**